src/cmd/moby/run_qemu.go

```diff
@@ -73,18 +91,38 @@ func runQemu(args []string) {
 	} else {
 		qemuArgs = append(qemuArgs, "-enable-kvm")
 		qemuArgs = append(qemuArgs, "-machine", "q35,accel=kvm:tcg")
+		if container {
+			dockerArgs = append(dockerArgs, "", "")
+			copy(dockerArgs[5:], dockerArgs[3:])
+			dockerArgs[3] = "--device"
+			dockerArgs[4] = "/dev/kvm"
+		}
+	}
+
+	var qemuCmd *exec.Cmd
+	var qemuImgCmd *exec.Cmd
+	if container {
+		dockerPath, err := exec.LookPath("docker")
+		if err != nil {
+			log.Fatal("Unable to find docker in the $PATH")
+		}
+		binArgs := append(dockerArgs, qemuBinPath)
+		imgArgs := append(dockerArgs, qemuImgPath)
+		qemuCmd = exec.Command(dockerPath, binArgs...)
+		qemuImgCmd = exec.Command(dockerPath, imgArgs...)
+	} else {
+		qemuCmd = exec.Command(fullQemuPath)
+		qemuImgCmd = exec.Command(fullQemuImgPath)
```

Comment:
the conditional above is for `container && /dev/kvm is present` and adds the necessary `--device /dev/kvm` runes to the docker run command. this conditional sets up the `*exec.Command` for both qemu and qemu image so the same set of args (built from the flags supplied) can be used.

there are still cases where we need to check whether we're in a container or not:
e.g: lines 143 and 176
as some options aren't compatible with containers (e.g GUI) or the paths that are being tested for are inside the container itself (e.g the OVMF firmware)